deltacat/experimental/storage/iceberg/visitor.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from typing import Any
+
+from deltacat.storage.model.expression import Reference, Literal
+from deltacat.storage.model.expression.visitor import ExpressionVisitor
+from pyiceberg.expressions import (
+    And,
+    Or,
+    Not,
+    EqualTo,
+    NotEqualTo,
+    GreaterThan,
+    GreaterThanOrEqual,
+    LessThan,
+    LessThanOrEqual,
+    IsNull,
+    In,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+
+class IcebergExpressionVisitor(ExpressionVisitor[None, Any]):
+    """
+    Visitor that translates DeltaCAT expressions to PyIceberg expressions.
+    Similar to GlueExpressionVisitor but outputs PyIceberg expressions.
+    """
+
+    def visit_reference(self, expr: Reference, context=None) -> str:
+        return expr.field
+
+    def visit_literal(self, expr: Literal, context=None) -> Any:
+        # Convert PyArrow scalar to Python native type
+        return expr.value.as_py() if hasattr(expr.value, "as_py") else expr.value
```

Comment:
Instead of just looking for an `as_py()` method, it may be safer to just be more explicit about the type we expect the expression type to be:
```suggestion
        return expr.value.as_py() if isinstance(expr.value, pyarrow.Scalar) else expr.value
```